From: Jan Kara <jack@suse.cz>
Subject: mm: Fix DIF failures on ext3 filesystems
References: bsc#971030
Patch-mainline: v3.9
Git-commit: 1d1d1a767206fbe5d4c69493b7e6d2a8d08cc0a0

The backport of commit 1d1d1a767206 (mm: only enforce stable page writes if the
backing device requires it) missed the addition of wait_for_stable_page() into
filemap_page_mkwrite(). Thus a page could be written to via mmap even though IO
was in progress for the page leading to DIF failures. The error during
backporting likely happened because we backported patch adding
filemap_page_mkwrite() only later and the original backporter did not realize
the consequences of the omission.

Add the forgotten wait_for_stable_page() call to fix the issue.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 mm/filemap.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -1939,6 +1939,7 @@ int filemap_page_mkwrite(struct vm_area_
 		ret = VM_FAULT_NOPAGE;
 		goto out;
 	}
+	wait_for_stable_page(page);
 out:
 	return ret;
 }
